### PR TITLE
pkg/ansible: Adding ability to have a flag that turns off the owner ref.

### PR DESCRIPTION
**Pull request by shawn-hurley**
* This will be used by an operator author by overriding the entrypoint
  in there build/Dockerfile.

**Description of the change:**

**Motivation for the change:**

### CODE REVIEW

**Comment by shawn-hurley** [comment]
@jmrodri Can you also take a look at this when you get a chance

File: doc/ansible/dev/advanced_options.md [hunk 1]
```diff
@@ -35,3 +35,11 @@ Some features can be overridden per resource via an annotation on that CR. The o
 ### Runner Directory
 
 The ansible runner will keep information about the ansible run in the container.  This is located `/tmp/ansible-operator/runner/<group>/<version>/<kind>/<namespace>/<name>`. To learn more  about the runner directory you can read the [ansible-runner docs](https://ansible-runner.readthedocs.io/en/latest/index.html).
+
+## Turning off Dependent Watches and Owner Reference Injection
+
+As user may want to manage what they watch and the owner references themseleves. This means that your operator will need to understand how to clean up after itslef when your CR is deleted. To disable these features you will need to edit your `build/Dockerfile` to include the line below.
```
Comment (fabianvf) [inline doc/ansible/dev/advanced_options.md]: Maybe replace the first sentence with something more like `You may want to manage what your operator watches and the owner references themselves.`

`s/itslef/itself/`

`s/line below./line below:`

Comment (fabianvf) [reply doc/ansible/dev/advanced_options.md]: Also I would say more explicitly that you cannot turn off ownerReferences without disabling dependent watches. Also perhaps describe what would happen if your watches file set `watchDependentResources: true` and inject owner references was disabled  (is it an error? Does the dependent watch just silently not happen?)

File: doc/ansible/dev/advanced_options.md [hunk 2]
```diff
@@ -35,3 +35,13 @@ Some features can be overridden per resource via an annotation on that CR. The o
 ### Runner Directory
 
 The ansible runner will keep information about the ansible run in the container.  This is located `/tmp/ansible-operator/runner/<group>/<version>/<kind>/<namespace>/<name>`. To learn more  about the runner directory you can read the [ansible-runner docs](https://ansible-runner.readthedocs.io/en/latest/index.html).
+
+## Turning off Dependent Watches and Owner Reference Injection
+
+You may want to manage what your operator watches and the owner references. This means that your operator will need to understand how to clean up after itslef when your CR is deleted. To disable these features you will need to edit your `build/Dockerfile` to include the line below.
```
Comment (fabianvf) [inline doc/ansible/dev/advanced_options.md]: s/itslef/itself

File: pkg/ansible/flags/flag.go [hunk 3]
```diff
@@ -31,5 +34,10 @@ func AddTo(flagSet *pflag.FlagSet, helpTextPrefix ...string) *AnsibleOperatorFla
 	aof := &AnsibleOperatorFlags{}
 	aof.WatchFlags.AddTo(flagSet, helpTextPrefix...)
 	flagSet.AddFlagSet(zap.FlagSet())
+	flagSet.BoolVar(&aof.InjectOwnerRef,
+		"inject-owner-ref",
+		true,
+		strings.Join(append(helpTextPrefix, "The ansible operator will inject owner reference unless this flag is false"), " "),
```
Comment (fabianvf) [inline pkg/ansible/flags/flag.go]: owner reference -> owner references?

File: doc/ansible/dev/advanced_options.md [hunk 4]
```diff
@@ -35,3 +35,13 @@ Some features can be overridden per resource via an annotation on that CR. The o
...
+
+**NOTE**: That if you use the feature there will be a warning that dependent watches is turnned off but there will be no error.
```
Comment (jmrodri) [inline doc/ansible/dev/advanced_options.md]: s/the feature/this feature/
s/turnned off/turned off/

File: images/scorecard-proxy/cmd/proxy/main.go [hunk 5]
```diff
@@ -60,7 +60,7 @@ func main() {
 		KubeConfig:        mgr.GetConfig(),
 		RESTMapper:        mgr.GetRESTMapper(),
 		ControllerMap:     cMap,
-		NoOwnerInjection:  true,
+		OwnerInjection:    false,
```
Comment (jmrodri) [inline images/scorecard-proxy/cmd/proxy/main.go]: Thanks for using the positive form!

File: pkg/ansible/proxy/proxy.go [hunk 6]
```diff
@@ -369,14 +374,18 @@ func Run(done chan error, o Options) error {
 		o.Cache = informerCache
 	}
 
-	if !o.NoOwnerInjection {
+	server.Handler = removeAuthorizationHeader(server.Handler)
+
+	if o.OwnerInjection {
 		server.Handler = InjectOwnerReferenceHandler(server.Handler, o.ControllerMap, o.RESTMapper, watchedNamespaceMap)
+	} else {
+		log.Info("Warning injection of owner references and dependent watches is turned off")
```
Comment (jmrodri) [inline pkg/ansible/proxy/proxy.go]: I'd add a colon after `Warning`

**Pull request merged**